src/main/kotlin/com/jetbrains/snakecharm/inspections/SmkSectionUnexpectedCallableArgsInspection.kt

```diff
@@ -0,0 +1,52 @@
+package com.jetbrains.snakecharm.inspections
```

Comment:
Please test your feature more attentively, at the moment doesn't work properly:

```
def params_function(wildcards, input, output, threads, resources):
    return "text"

MSG = "msg"
def conda_env(arg):
    return str(arg)

rule foo:
    conda: conda_env # here should be error
    output: MSG # here should be no error, MSG is string, not calable
    benchmark: conda_env("arg") 
    message: MSG   # here should be no error, MSG is string, not calable
    log: params_function  # here should be error
```

![image](https://user-images.githubusercontent.com/72933/89877932-81c06300-dbc9-11ea-9dfe-980e89f9db9c.png)

Reply:
I found no documentation about which sections accept callable arguments and tested it manually. It seems like in first commit including 'log' section was excess, I removed it now.
Also fixed other references false-positive warning, now it applies only to functions and lambdas.

Reply:
>  But section 'conda' works fine with callable, when I try it locally.
Do you launch snakemake with `--use-conda`? Otherwise, conda sections are just ignored.

Reply:
> I found no documentation about which sections accept callable arguments and tested it manually.

Likely it is only in snakemake sources

Reply:
Callable in `conda` doesn't work without/with `--use-conda`
![image](https://user-images.githubusercontent.com/72933/90031398-d943f900-dcc5-11ea-819a-6eadccff0253.png)

Reply:
> which sections accept callable arguments and tested it manually

As for the list of valid/invalid section names, please see #198 and related source code investigation in #187. Also in snakemake 5.16.0 see the method  `Workflow.rule.decorate` (snakemake/workflow.py:1062). it provides assertions for different section types, e.g. direct check that `theads` and `resources` cannot be a callable. So I believe your test isn't correct.

Reply:
Found checkpoints for callable arguments in source code of snakemake, now PR fixed according to it.

Reply:
See [comment](https://github.com/JetBrains-Research/snakecharm/issues/187#issuecomment-674038509) in #187 discussion for extra information.